### ISSUE TITLE
The 4W4A LLaMa2 results on  six zero-shot datasets

### ISSUE THREAD

**Issue by XA23i** [bytedance/AffineQuant#3]
Hi, I notice in your paper there are only 4W4A LLaMa1 results on  six zero-shot datasets. May i ask do you have 4W4A LLaMa2 results on zero-shot tasks. Thanks.

**Comment by bobma-bytedance** [comment]
> Hi, I notice in your paper there are only 4W4A LLaMa1 results on six zero-shot datasets. May i ask do you have 4W4A LLaMa2 results on zero-shot tasks. Thanks.

Hi XA23i,

  We did not perform W4A4 zero-shot quantization on LLaMA2. If you have the computing resources, you can try the W4A4 configuration. For the running scripts, you can directly use the W4A4 running scripts for LLaMA2 in OmniQuant ([scripts](https://github.com/OpenGVLab/OmniQuant/blob/main/scripts/Llama-2/Llama-2-7b/w4a4.sh)).

  If you have any questions, feel free to ask.

Best regards.

**Comment by XA23i** [comment]
Hi, i run the scripts below to evaluate llama2-7B 
CUDA_VISIBLE_DEVICES=0 python main.py \
--model /path/to/Llama-2-7b-hf --eval_ppl --save_dir ./fake_quant_model/Llama-2-7b-w4a4 \
--epochs 20 --output_dir ./log/Llama-2-7b-w4a4 --act-scales ./act_scales/Llama-2-7b.pt --act-shifts ./act_shifts/Llama-2-7b.pt \
--wbits 4 --abits 4 --lwc --let --use_matrix --sf 0.1 --let_lr 1e-3 --alpha 0.75 \
--taskspiqa,arc_easy,arc_challenge,boolq,hellaswag,winogrande

However, the ppl of wikitext and c4 are 13.59 and 18.17, which is different with 12.69  and 15.76 in your paper.
Do I set the wrong hyperparameters or any other tricks?

**Comment by bobma-bytedance** [comment]
> Hi, i run the scripts below to evaluate llama2-7B CUDA_VISIBLE_DEVICES=0 python main.py --model /path/to/Llama-2-7b-hf --eval_ppl --save_dir ./fake_quant_model/Llama-2-7b-w4a4 --epochs 20 --output_dir ./log/Llama-2-7b-w4a4 --act-scales ./act_scales/Llama-2-7b.pt --act-shifts ./act_shifts/Llama-2-7b.pt --wbits 4 --abits 4 --lwc --let --use_matrix --sf 0.1 --let_lr 1e-3 --alpha 0.75 --taskspiqa,arc_easy,arc_challenge,boolq,hellaswag,winogrande
> 
> However, the ppl of wikitext and c4 are 13.59 and 18.17, which is different with 12.69 and 15.76 in your paper. Do I set the wrong hyperparameters or any other tricks?

I have provided our training log, where you can compare hyperparameters and the changes in loss, among other information. Additionally, it includes the zero-shot dataset accuracy for LLaMA2.

We previously obtained results for LLaMA2 w4a4, but they were not included in the paper. I have forgotten it.  :)

Best regards.

[log_rank0_1706942168.txt](https://github.com/user-attachments/files/16541287/log_rank0_1706942168.txt)
